File: eWallet/app/src/main/java/com/mob/welups/ui/splash/SplashActivity.java
package com.mob.welups.ui.splash;

import android.app.Activity;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.res.Configuration;
import android.os.Handler;
import android.os.Looper;

import androidx.lifecycle.ViewModelProvider;

import com.mob.welups.base.v2.BaseActivityV2;
import com.mob.welups.config.LocaleHelper;
import com.mob.welups.databinding.ActionBarBinding;
import com.mob.welups.databinding.ActivitySplashBinding;
import com.mob.welups.ui.onboarding.OnBoardingActivity;

import java.util.Locale;

public class SplashActivity extends BaseActivityV2<ActivitySplashBinding, SplashViewModel> {

    private final Handler handler = new Handler(Looper.getMainLooper());
    private final Runnable runnable = new Runnable() {
        @Override
        public void run() {
            startActivity(new Intent(SplashActivity.this, OnBoardingActivity.class));
            finish();
        }
    };

    @Override
    public ActivitySplashBinding createViewBinding() {
        return ActivitySplashBinding.inflate(getLayoutInflater());
    }

    @Override
    public ActionBarBinding createActionBarViewBinding() {
        return null;
    }

    @Override
    public SplashViewModel createViewModel() {
        return new ViewModelProvider(this).get(SplashViewModel.class);
    }

    @Override
    public void initialize() {
        setLanguage();

        hideItemActionBar();
        hideLogoWelupsAndMenu();

        handler.postDelayed(runnable, 3000);
    }

    private void setLanguage() {
        SharedPreferences prefs = getSharedPreferences("CommonPrefs", Activity.MODE_PRIVATE);
        LocaleHelper.language = prefs.getString("Language", LocaleHelper.LOCALE_VI_VN);
        Locale locale = new Locale(LocaleHelper.language);
        Locale.setDefault(locale);
        Configuration config = new Configuration();
        config.locale = locale;
        getBaseContext().getResources().updateConfiguration(config,
                getBaseContext().getResources().getDisplayMetrics());
    }
}

File: Apps/src/main/java/com/android/container/splash/SplashActivity.kt
package com.android.container.splash

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Handler
import android.os.Looper
import androidx.lifecycle.ViewModelProvider
import com.android.base.BaseActivity
import com.android.container.MainActivity
import com.android.databinding.ActivitySplashBinding

@SuppressLint("CustomSplashScreen")
class SplashActivity : BaseActivity<ActivitySplashBinding, SplashViewModel>() {
    override fun createViewBinding(): ActivitySplashBinding {
        return ActivitySplashBinding.inflate(layoutInflater)
    }

    override fun createViewModel(): SplashViewModel {
        return ViewModelProvider(this)[SplashViewModel::class.java]
    }

    override fun initializeViews() {
        val handler = Handler(Looper.getMainLooper())
        val runnable = Runnable {
            val intent = Intent(applicationContext, MainActivity::class.java)
            startActivity(intent)
        }
        handler.postDelayed(runnable, 1500)
    }

    override fun initializeEvents() {}
}